HPXMLtoOpenStudio/resources/hvac.rb

```diff
@@ -2949,78 +2996,100 @@ def self.calculate_odb_at_zero_cop_or_capacity(data, user_odbs, property, find_h
     end
   end
 
-  # TODO
+  # Extrapolates the given performance property for the specified target value and property.
   #
-  # @param detailed_performance_data [TODO] TODO
-  # @param capacity_description [TODO] TODO
-  # @param target_odb [TODO] TODO
-  # @param property [TODO] TODO
-  # @return [TODO] TODO
-  def self.interpolate_to_odb_table_point(detailed_performance_data, capacity_description, target_odb, property)
-    data = detailed_performance_data.select { |dp| dp.capacity_description == capacity_description }
-
-    target_dp = data.find { |dp| dp.outdoor_temperature == target_odb }
+  # @param datapoints [HPXML::CoolingDetailedPerformanceData or HPXML::HeatingDetailedPerformanceData] Array of detailed performance datapoints at a given speed
+  # @param capacity_description [String] The capacity description (HPXML::CapacityDescriptionXXX)
+  # @param target_property [Symbol] The datapoint property for the target value (e.g., :outdoor_temperature)
+  # @param target_value [Double] The target value to extrapolate to (F)
+  # @param property [Symbol] The datapoint property to extrapolate (e.g., :capacity, :efficiency_cop, etc.)
+  # @param slope_requirement [Symbol] The slope requirement (:positive or :negative)
+  # @return [Double] The extrapolated value (F)
+  def self.extrapolate_data_point(datapoints, capacity_description, target_property, target_value, property, slope_requirement = nil)
+    datapoints = datapoints.select { |dp| dp.capacity_description == capacity_description }
```

Comment:
Can we rename this variable to avoid modifying argument? Or is it intended?

Reply:
It may look a little weird but I'd say it's intended. Any calls from `hvac.rb` already pass in datapoints that have been filtered by the CapacityDescription, but calls from `defaults.rb` and `hvac_sizing.rb` pass in the full datapoints list and need to be filtered. The code in this method should always operate on the filtered list of datapoints. Open to suggestions though...

Reply:
Oh wait, I think I understand your concern now. It doesn't actually modify the array that the calling method passed in.